website/docs/getting-started/setup/instance-configuration/google-maps.mdx

```diff
@@ -0,0 +1,69 @@
+---
+description: Follow these steps to configure Google Maps Key for your self-hosted Appsmith instance.
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Google Maps
+
+This page provides instructions to integrate Google Maps to use the [Map Widget](/reference/widgets/maps) in your applications on the self-hosted instance.
+
+## Prerequisites
+
+- A [Google Cloud account](https://www.google.com/aclk?sa=l&ai=DChcSEwi0373S7OX_AhW7Vn0KHT5BD30YABABGgJzZg&sig=AOD64_3YNMGHJAu3vGJmjK-vrl8qni2BxA&q&adurl&ved=2ahUKEwjjs7XS7OX_AhWy3TgGHWtHAkYQ0Qx6BAgIEAE). 
+
+## Setup Google Maps platform
+
+1. Log into the [Google Cloud console](https://console.cloud.google.com/).
+2. Generate a Google Maps API key. Follow the [Create API keys](https://developers.google.com/maps/documentation/javascript/get-api-key#create-api-keys) guide on the official Google Maps platform documentation.
```

Comment:
I think we forgot to instruct to enable the APIs

Maps Embed API
Maps JavaScript API
Places API
Geolocation API

Reply:
The restrict API key section covers that so haven't added that as a separate point.

Reply:
does it ask to enable all the above four specifically?

Reply:
Based on what users are using they can add all the APIs so that it's available for them on Maps platform. And it has the four APIs listed too 
>Select Maps JavaScript API from Select APIs dropdown. If the Maps JavaScript API is not listed, you need to [enable](https://developers.google.com/maps/documentation/javascript/cloud-setup#enabling-apis) it.
If your project uses Places Library, also select Places API. Similarly, if your project uses other services in the JavaScript API ([Directions Service](https://developers.google.com/maps/documentation/javascript/directions), [Distance Matrix Service](https://developers.google.com/maps/documentation/javascript/distancematrix), [Elevation Service](https://developers.google.com/maps/documentation/javascript/elevation), and/or [Geocoding Service](https://developers.google.com/maps/documentation/javascript/geocoding)), you must also enable and select the corresponding API in this list.

Reply:
I think we have to specify that these API need to be enabled. It can also be in the prerequisites section

Reply:
If they don't enable either one of them, for example geolocation, some things at our end may not work

Reply:
Though it shouldn't as whatever you select those API functions you should be able to use. However, as I can't test it.  I've added it as a fourth point to the generate api key section.